### PR TITLE
store/tikv: fix goroutine leak in gcworker

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Fix goroutine leak:

```
goroutine 28938475 [chan send, 485 minutes]:
github.com/pingcap/tidb/store/tikv.(*RangeTaskRunner).RunOnRange(0xc014fa6f40, 0x21eed60, 0xc00b3592c0, 0x3483a58, 0x0, 0x0, 0x3483a58, 0x0, 0x0, 0x0, ...)
/home/jenkins/workspace/build_tidb_release-3.0/go/src/github.com/pingcap/tidb/store/tikv/range_task.go:171 +0xd18
github.com/pingcap/tidb/store/tikv/gcworker.(*GCWorker).resolveLocks(0xc00b356e00, 0x21eed60, 0xc00b3592c0, 0x5abe9147da00000, 0x3, 0xc003fb53c0, 0x90)
/home/jenkins/workspace/build_tidb_release-3.0/go/src/github.com/pingcap/tidb/store/tikv/gcworker/gc_worker.go:691 +0x58d
github.com/pingcap/tidb/store/tikv/gcworker.(*GCWorker).runGCJob(0xc00b356e00, 0x21eed60, 0xc00b3592c0, 0x5abe9147da00000, 0x3)
/home/jenkins/workspace/build_tidb_release-3.0/go/src/github.com/pingcap/tidb/store/tikv/gcworker/gc_worker.go:425 +0xd8
created by github.com/pingcap/tidb/store/tikv/gcworker.(*GCWorker).leaderTick
/home/jenkins/workspace/build_tidb_release-3.0/go/src/github.com/pingcap/tidb/store/tikv/gcworker/gc_worker.go:288 +0xc6b
```

### What is changed and how it works?

In range task runner's `RunOnRange` function, it create many workers, and dispatch task to the workers through the `taskCh`.

When the worker meets the first error, it call `cancel`, which makes all workers exit, but the dispatch loop is still writing to the channel... so `RunOnRange` blocks on here `taskCh <- task` forever.

In this commit I make the sender check `<-ctx.Done()` to handle the error case

### Check List 

Tests 

 - Integration test
Our schrodinger test platform find this leak and it's easy to reproduce there.

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @MyonKeminta @disksing

**Comment by tiancaiamao** [comment]
PTAL @MyonKeminta @disksing

**Comment by MyonKeminta** [comment]
@disksing PTAL

File: store/tikv/range_task.go [hunk 1]
```diff
@@ -168,7 +166,12 @@ func (s *RangeTaskRunner) RunOnRange(ctx context.Context, startKey []byte, endKe
 		}
 
 		pushTaskStartTime := time.Now()
-		taskCh <- task
+
+		select {
+		case taskCh <- task:
+		case <-ctx.Done():
+			break
```
Comment (disksing) [inline store/tikv/range_task.go]: I think need return err here.

Comment (tiancaiamao) [reply store/tikv/range_task.go]: It should run to line https://github.com/pingcap/tidb/pull/10622/files#diff-8a388eccc19df74aa4991690c5f508a5L192 to return error!

Comment (disksing) [reply store/tikv/range_task.go]: got it :) LGTM!

**Pull request merged**